src/test/store/profile-view.test.js

```diff
@@ -216,14 +217,20 @@ describe('getJankMarkersForHeader', function() {
   });
 
   it('will create a jank instance with eventDelay values', function() {
-    const breakingPoint = 70;
-    const eventDelay = [0, 20, 40, 60, breakingPoint, 0, 20, 40];
+    const breakingPoint = 50;
+    // We need to fill the samples before, because the new algorithm is different
+    // and the numbers usually means that we have a delay before that time.
+    const eventDelay = Array(50).fill(0);
+    eventDelay.push(5, 10, 15, 25, 30, 40, breakingPoint, 1);
     const jankInstances = setupWithEventDelay({
       sampleCount: eventDelay.length,
       eventDelay,
     });
     expect(jankInstances.length).toEqual(1);
-    expect(jankInstances[0].dur).toEqual(breakingPoint);
+    // This is breakingPoint + 2 because that means that the jank actually
+    // started two samples before that breaking point, this is done by the
+    // pre-processing.
+    expect(jankInstances[0].dur).toEqual(breakingPoint + 2);
```

Comment:
It would be interesting to test the `start` as well.

Also I admit I don't understand the logic, even with the comment. Maybe adding a bigger comment with some graphs would help...